gitpod-io/gitpod
# Introduce prebuild task tab component

## Description

Cleans up our task tab solution by introducing a `<PrebuildTaskTab>` component, allowing us to properly manage per-tab event listeners and xterm instances.

We now cache all logs of all tabs in-memory, so switching tabs is instantenious.

## Related Issue(s)

Fixes ENT-279

## How to test

https://ft-prebuila62e51af40.preview.gitpod-dev.com/workspaces

1. Go to import a repo and start a prebuild
2. Make sure you can tab through  them and logs persist

## Documentation


#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
